bip-0340.mediawiki

```diff
@@ -213,6 +213,49 @@ The algorithm ''BatchVerify(pk<sub>1..u</sub>, m<sub>1..u</sub>, sig<sub>1..u</s
 
 If all individual signatures are valid (i.e., ''Verify'' would return success for them), ''BatchVerify'' will always return success. If at least one signature is invalid, ''BatchVerify'' will return success with at most a negligible probability.
 
+=== Usage Considerations ===
+
+==== Messages of Arbitrary Size ====
+
+The signature scheme specified in this BIP accepts byte strings of arbitrary size as input messages.<ref>In theory, the message size is restricted due to the fact that SHA256 accepts byte strings only up to size of 2^56-1 bytes.</ref>
```

Comment:
Where does the number 2^56-1 come from?

Reply:
SHA256 accepts messages of length up to 2^64-1 *bits*, see https://nvlpubs.nist.gov/nistpubs/FIPS/NIST.FIPS.180-4.pdf Figure 1.

Reply:
Ah right, tricky.

Reply:
2^64 bits is 2^61 bytes

Reply:
Ah right, tricky... Sigh.

Reply:
fixed